src/rfc3986/abnf_regexp.py

```diff
@@ -88,51 +88,50 @@
 # about ABNF (Augmented Backus-Naur Form) use in the comments
 variations = [
     #                            6( h16 ":" ) ls32
-    '(%(hex)s:){6}%(ls32)s' % _subs,
+    '(?:%(hex)s:){6}%(ls32)s' % _subs,
     #                       "::" 5( h16 ":" ) ls32
-    '::(%(hex)s:){5}%(ls32)s' % _subs,
+    '::(?:%(hex)s:){5}%(ls32)s' % _subs,
     # [               h16 ] "::" 4( h16 ":" ) ls32
-    '(%(hex)s)?::(%(hex)s:){4}%(ls32)s' % _subs,
+    '(?:%(hex)s)?::(?:%(hex)s:){4}%(ls32)s' % _subs,
     # [ *1( h16 ":" ) h16 ] "::" 3( h16 ":" ) ls32
-    '((%(hex)s:)?%(hex)s)?::(%(hex)s:){3}%(ls32)s' % _subs,
+    '(?:(%(hex)s:)?%(hex)s)?::(?:%(hex)s:){3}%(ls32)s' % _subs,
     # [ *2( h16 ":" ) h16 ] "::" 2( h16 ":" ) ls32
-    '((%(hex)s:){0,2}%(hex)s)?::(%(hex)s:){2}%(ls32)s' % _subs,
+    '(?:(%(hex)s:){0,2}%(hex)s)?::(?:%(hex)s:){2}%(ls32)s' % _subs,
     # [ *3( h16 ":" ) h16 ] "::"    h16 ":"   ls32
-    '((%(hex)s:){0,3}%(hex)s)?::%(hex)s:%(ls32)s' % _subs,
+    '(?:(%(hex)s:){0,3}%(hex)s)?::%(hex)s:%(ls32)s' % _subs,
     # [ *4( h16 ":" ) h16 ] "::"              ls32
-    '((%(hex)s:){0,4}%(hex)s)?::%(ls32)s' % _subs,
+    '(?:(%(hex)s:){0,4}%(hex)s)?::%(ls32)s' % _subs,
     # [ *5( h16 ":" ) h16 ] "::"              h16
-    '((%(hex)s:){0,5}%(hex)s)?::%(hex)s' % _subs,
+    '(?:(%(hex)s:){0,5}%(hex)s)?::%(hex)s' % _subs,
     # [ *6( h16 ":" ) h16 ] "::"
-    '((%(hex)s:){0,6}%(hex)s)?::' % _subs,
+    '(?:(%(hex)s:){0,6}%(hex)s)?::' % _subs,
 ]
 
-IPv6_RE = '(({0})|({1})|({2})|({3})|({4})|({5})|({6})|({7})|({8}))'.format(
+IPv6_RE = '((?:{0})|(?:{1})|(?:{2})|(?:{3})|(?:{4})|(?:{5})|(?:{6})|(?:{7})|(?:{8}))'.format(
     *variations
 )
 
+# RFC 6874 Zone ID ABNF
+ZONE_ID = '(?:[' + UNRESERVED_RE + ']|' + PCT_ENCODED + ')+'
+IPv6_WITH_ZONES_RE = IPv6_RE + '(?:(?:%25|%)' + ZONE_ID + ')?'
```

Comment:
Two separate patterns here, one for the validator and one for the parser.

Reply:
This module is a public API iirc